cmake/sdlchecks.cmake

```diff
@@ -30,28 +30,19 @@ macro(FindLibraryAndSONAME _LIB)
 endmacro()
 
 macro(CheckDLOPEN)
-  check_symbol_exists(dlopen "dlfcn.h" HAVE_DLOPEN)
-  if(NOT HAVE_DLOPEN)
-    check_library_exists(dl dlopen "" DLOPEN_LIB)
-    if(DLOPEN_LIB)
+  cmake_push_check_state(RESET)
+  check_symbol_exists(dlopen "dlfcn.h" HAVE_DLOPEN_IN_DLFCN_H_1)
```

Comment:
`HAVE_DLOPEN_IN_LIBC` looks better for this

Reply:
Your suggestions sounded way better, so I applied them.